File: library/src/main/java/org/xmtp/android/library/libxmtp/Member.kt
package org.xmtp.android.library.libxmtp

import org.xmtp.android.library.ConsentState
import org.xmtp.android.library.InboxId
import uniffi.xmtpv3.FfiConversationMember
import uniffi.xmtpv3.FfiPermissionLevel

enum class PermissionLevel {
    MEMBER, ADMIN, SUPER_ADMIN
}
class Member(private val ffiMember: FfiConversationMember) {

    val inboxId: InboxId
        get() = ffiMember.inboxId
    val identities: List<PublicIdentity>
        get() = ffiMember.accountIdentifiers.map { PublicIdentity(it) }
    val permissionLevel: PermissionLevel
        get() = when (ffiMember.permissionLevel) {
            FfiPermissionLevel.MEMBER -> PermissionLevel.MEMBER
            FfiPermissionLevel.ADMIN -> PermissionLevel.ADMIN
            FfiPermissionLevel.SUPER_ADMIN -> PermissionLevel.SUPER_ADMIN
        }

    val consentState: ConsentState
        get() = ConsentState.fromFfiConsentState(ffiMember.consentState)
}

File: sdks/ios/Sources/XMTPiOS/Libxmtp/Member.swift
//
//  Member.swift
//
//
//  Created by Naomi Plasterer on 5/27/24.
//

import Foundation

public enum PermissionLevel {
	case Member, Admin, SuperAdmin
}

public struct Member {
	var ffiGroupMember: FfiConversationMember

	public var inboxId: InboxId {
		ffiGroupMember.inboxId
	}

	public var identities: [PublicIdentity] {
		ffiGroupMember.accountIdentifiers.map { PublicIdentity(ffiPrivate: $0) }
	}

	public var permissionLevel: PermissionLevel {
		switch ffiGroupMember.permissionLevel {
		case .member:
			PermissionLevel.Member
		case .admin:
			PermissionLevel.Admin
		case .superAdmin:
			PermissionLevel.SuperAdmin
		}
	}

	public var consentState: ConsentState {
		ffiGroupMember.consentState.fromFFI
	}
}
